src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs

```diff
@@ -988,22 +987,29 @@ private void WriteString(in DbRow row, Utf8JsonWriter writer)
                     arrayItemsCount = row.SizeOrLength;
                     numberOfRowsForValues += row.NumberOfRows;
                 }
-                else if (tokenType == JsonTokenType.PropertyName)
+                else if (IsStringLike(tokenType)) // Equivalent to (tokenType == JsonTokenType.PropertyName || tokenType == JsonTokenType.String)
                 {
                     numberOfRowsForValues++;
                     numberOfRowsForMembers++;
-                    database.Append(tokenType, tokenStart, reader.ValueSpan.Length);
+                    Debug.Assert(reader.TokenStartIndex < int.MaxValue); // Adding 1 to skip the start quote will never overflow
+                    database.Append(tokenType, tokenStart + 1, reader.ValueSpan.Length);
+
+                    Debug.Assert((tokenType == JsonTokenType.PropertyName && !inArray) || tokenType == JsonTokenType.String);
+
+                    if (inArray)
+                    {
+                        Debug.Assert(tokenType == JsonTokenType.String);
+                        arrayItemsCount++;
```

Comment:
Previous bug?

Reply:
Nope, this is just an artifact of moving the "string" case up with the "property name" case and merging them here.

If you look, previously, we had this in the string case already.